actionpack/lib/action_dispatch/system_testing/driver.rb

```diff
@@ -9,23 +9,43 @@ def initialize(name, **options)
       end
 
       def use
-        register if selenium?
+        case @name
+        when :selenium then register_selenium
+        when :poltergeist then register_poltergeist
+        when :webkit then register_webkit
+        end
+
         setup
       end
 
       private
-        def selenium?
-          @name == :selenium
-        end
-
-        def register
+        def register_selenium
           Capybara.register_driver @name do |app|
             Capybara::Selenium::Driver.new(app, { browser: @browser }.merge(@options)).tap do |driver|
               driver.browser.manage.window.size = Selenium::WebDriver::Dimension.new(*@screen_size)
             end
           end
         end
 
+        def register_poltergeist
+          if defined?(Capybara::Poltergeist)
```

Comment:
What is the `if defined?(Capybara::Poltergeist)` buying us? This won't load without register getting called and if you don't add the gemfile it will blow up anyway. Otherwise this will silently fail I think.